### PR TITLE
fix: mongodb-redact contains slow regexes MONGOSH-1392

### PR DESCRIPTION
**Pull request by alenakhineika**
Replace a base64 string of `BinData` with a placeholder to ensure the rest of the regex list is not being run and stuck on a large blob of the input.

### CODE REVIEW

**Review by mcasimir — APPROVED** [review]

File: lib/regexes.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 var regexes = [
+  // BinData excluding 3 and 4 subtypes which are UUID and should not be long
+  [/(BinData\()((0|1|2|5|128))(\s*,\s*['"]?)[A-Za-z0-9+/=\s]+(['"]?\))/mg, '$1$2$4<base64>$5'],
```
Comment (addaleax) [inline lib/regexes.js:3]: What about other subtypes than 0,1,2,3,4,5,128? Would it hurt to allow any number (up to three digits, I guess) here?

**Review by addaleax — COMMENTED** [review]
Clever! I think this is a great addition for this package. We'll need to be careful not to close MONGOSH-1392, though, because the issue described in it doesn't apply to `BinData` usage specifically.

File: lib/regexes.js (hunk 1 above)
Comment (alenakhineika) [reply lib/regexes.js:3]: I didn't want to reduce too much :) but if you think this is ok we can allow any number!

**Review by addaleax — APPROVED** [review]
Changes in this PR look good!

**Pull request merged**